geomesa-index-api/src/main/scala/org/locationtech/geomesa/index/planning/guard/GraduatedQueryGuard.scala

```diff
@@ -8,76 +8,61 @@
 
-  override def guard(strategy: QueryStrategy): Option[IllegalArgumentException] = {
-    FullTableScanQueryGuard.guard(strategy) match {
-      case Some(illegalArgumentException) => Some(illegalArgumentException)
-      case None => guardInner(strategy)
+  override def init(ds: DataStore, sft: SimpleFeatureType): Unit = {
+    disabled = GraduatedQueryGuard.disabled(sft.getTypeName)
+    if (disabled) {
+      logger.info(s"This guard is disabled for schema '${sft.getTypeName}' via system property")
     }
+    // let any errors bubble up and disable this guard
+    guardLimits = buildLimits(ConfigFactory.load().getConfigList(s"$ConfigPath.${sft.getTypeName}"))
   }
 
-  private def guardInner(strategy: QueryStrategy): Option[IllegalArgumentException] = {
-    val intervalsOption: Option[FilterValues[Bounds[ZonedDateTime]]] = strategy.values.collect { case v: TemporalIndexValues => v.intervals }
-    val spatialBoundsOption: Option[Seq[(Double, Double, Double, Double)]] = strategy.values.collect { case v: SpatialIndexValues => v.spatialBounds }
-
-    val spatialExtent = spatialBoundsOption match {
-      case None => 360.0 * 180.0  // Whole world
-      case Some(seq) => seq.map { case  (lx: Double, ly: Double, ux: Double, uy: Double) =>
-        (ux-lx) * (uy-ly)
-      }.sum
-    }
+  override def rewrite(query: Query): Unit = {}
 
-    intervalsOption match {
-      // The None case reflects a query without a temporal extent or a full-table scan on a table with a temporal extent
-      // The full-table scan block should be used to avoid the later case.
-      // Here a none is returned indicating that this guard does not stop the query
-      case None => None
-      case Some(intervals) =>
-        if (!intervals.forall(_.isBoundedBothSides) || intervals.isEmpty) {
-          Some(new IllegalArgumentException("At least one part of the query is temporally unbounded.  Add bounded temporal constraints."))
-        } else {
-          val queryDuration = duration(intervals.values)
-          guardLimits.find( _.sizeLimit >= spatialExtent).flatMap {
-            limit =>
-              if (queryDuration > limit.durationLimit) {
-                Some(new IllegalArgumentException(
-                  s"Query of this size exceeds maximum allowed filter duration of ${limit.durationLimit}: " +
-                    s"${filterToString(strategy.filter.filter)}. " +
-                    s"Try a small spatial extent or a shorter time range."))
-              } else {
-                None
-              }
+  override def guard(strategy: QueryStrategy): Option[IllegalArgumentException] = {
+    val msg = if (disabled || !strategy.index.isInstanceOf[SpatioTemporalIndex[_, _]]) { None } else {
+      val values = strategy.values.collect {
+        case v: SpatialIndexValues with TemporalIndexValues => (v.spatialBounds, v.intervals)
+      }
+      values match {
+        case None => Some("Query does not have a temporal filter")
```

Comment:
The query could be lacking a spatial filter, no?  E.g., wouldn't this be hit when you pass "dtg during date1/date2" to a Z3 index?

Reply:
i think this would only be None if there is no spatial AND no temporal filter. since it's a temporal guard, i was thinking the error message should reference the temporal part only. what do you think?

Reply:
I think of the graduate guard as a spatial temporal guard; that's why I'm asking about it. 

It seems like the other errors show what filter limits are allowed.